vendor/github.com/weaveworks/common/server/server.go

```diff
@@ -31,6 +31,16 @@ import (
 	"github.com/weaveworks/common/signals"
 )
 
+// SignalHandler used by Server.
+type SignalHandler interface {
```

Comment:
This is going to be merged by PR https://github.com/cortexproject/cortex/pull/2576. I suggest merge them in order.